CMakeLists.txt

```diff
@@ -107,10 +108,16 @@ if (NOT CMAKE_BUILD_TYPE)
   message ("No build type detected. Build type set to Release by default")
   set ( CMAKE_BUILD_TYPE Release )
 endif()
-
-set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-rtti -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DVC_DEBUG -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -fPIC -fvisibility-inlines-hidden" )
+set ( CMAKE_CXX_STANDARD 14 )
+set ( CMAKE_CXX_STANDARD_REQUIRED ON )
+set ( CMAKE_CXX_EXTENSIONS OFF )
+set ( CMAKE_VISIBILITY_INLINES_HIDDEN ON )
+set ( CMAKE_POSITION_INDEPENDENT_CODE ON )
+set ( CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)  
+set ( CMAKE_CXX_VISIBILITY_INLINES_HIDDEN ON )
+set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-use-cxa-atexit -fno-rtti" )
+set ( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DVC_DEBUG" )
```

Comment:
This one can be moved to source_file_property > compile flag, under conditional "Debug" CMAKE_BUILD_TYPE